### ISSUE TITLE
Add media player

### ISSUE THREAD

**Issue by alexdekok** [minvws/nl-covid19-notification-app-website#61]
Video content kunnen afspelen volgens richtlijnen Rijksoverheid

## Definition of done
- [x] The video player is accesssible according to [Dutch government WCAG standards for video](https://www.digitoegankelijk.nl/uitleg-van-eisen/geluid-en-video)
- [x] Lazy loaden van video en javascript
- [x] Subtitles in NL  & EN, switches with site language toggle. 

## Issue

Rijksoverheid faciliteert video maar ook audio en animaties voor iedereen. Of je nu blind, doof, slechtziend of in een andere situatie zit…wat we laten bewegen / vertellen of animeren moet voor iedereen te begrijpen zijn.
E.g. 

Op Rijksoverheid wordt video of audiocontent daarom aangeboden via de Mediatheek van Rijkshoverheid. 

## Op sites die binnen het eigen platform (Platform Rijksoverheid) draaien, wordt dit als volgt opgelost:

- Video’s vanuit de mediatheek (van Rijksoverheid) worden ingeladen en aangeboden op alle formaten die er nodig zijn om iedereen te kunnen bedienen.
- Voorbeeld: de video op deze pagina https://www.iedereenondereendak.nl/wat-er-al-gebeurt/wonen-met-begeleiding/optimaal-leven
- De video’s worden als volgt aangeleverd vanuit de mediatheek (voorbeeld):
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-bron.mp4
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-bron.mxf
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-web-hd.mp4
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-still-klein.jpg
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-still-middel.jpg
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-still-groot.jpg
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-audio.mp3
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-ondertiteling.srt
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-ondertiteling.vtt
www.rovid.nl/vws/dco/2020/vws-dco-20200622-idubbvnqy-tekst.txt

## Geen youtube / vimeo players
Belangrijk om te weten is dus dat YouTube/Vimeo-players dus niet gebruikt worden op Overheidssites. Dat betekent dat er dus een player geintegreerd moet worden die vanuit de overheidsmediatheek video/animaties en audio kan uitserveren.

**Comment by EdoPlantinga** [comment]
@mellewynia: You may want to consult the code for NL community for a suitable accessible player, @bwbroersma had some ideas too I believe.

**Comment by bwbroersma** [comment]
The [example referenced](https://www.iedereenondereendak.nl/wat-er-al-gebeurt/wonen-met-begeleiding/optimaal-leven) uses [MediaElementPlayer](http://www.mediaelementjs.com/), see [this minimal](https://github.com/mediaelement/mediaelement/blob/master/docs/usage.md#automatic-start) example.
Probably [video.js](https://github.com/videojs/video.js/) like mentioned in #28 is an option too.

**Comment by EdoPlantinga** [comment]
@bwbroersma suggested previously:
Probably do native <video> ➕ video.js add-on?

**Comment by arianvp** [comment]
Ik denk ik breng het even hier op deze issue op de radar. Op de huidige website van het Corona-app onderzoek wordt een Vimeo (met tracking) gebruikt. Wat mij niet de bedoeling lijkt.

https://www.onderzoektwente.nl/vragenlijst.html

**Comment by arianvp** [comment]
Aangezien het gaat om de website van het onderzoeksplatform dat gebruikt wordt zou denk ik  het vermelden van Vimeo als derde partij voldoende zou moeten zijn om dit op te lossen. Aangezien de video voor de rest geen third-party tracking cookies o.i.d. gebruikt; dat zit allemaal goed.

**Comment by dirkx** [comment]
Ack - we zijn hier al achteraan met alle urgentie, etc.. Zowel dit geval als 'hoe dit kon gebeuren'.

En ik ben het eens met je oplossing - maar hoop dat we een stuk beter kunnen doen.

**Comment by dirkx** [comment]
Video removed; post-mortem & remedial underway & governance/avg actions underway.

**Comment by alexdekok** [comment]
@EdoPlantinga @bwbroersma @mellewynia ik krijg zojuist door dat video afspelen vanuit de mediatheek van Rijksoverheid niet gaat lukken. Men verwacht de druk niet aan te kunnen qua hoeveelheden. 

2 vragen:

- Is er een alternatieve manier om video's te laten hosten? 
- Is er een alternatieve manier om toegankelijke video's aan te bieden?

Gr.

Alex

**Comment by alexdekok** [comment]
zojuist sprak ik @mellewynia ik heb wat research gedaan bij partijen waar ik eerder mee heb samengewerkt voor het ontwikkelen van externe Rijksoverheids websites. Zij maakten overigens als player gebruik van http://www.mediaelementjs.com/

**Comment by mellewynia** [comment]
@mdraijer, net even @mdraijer, en hij gaat de vraag van de hosting van de video voorleggen. Hopelijk hebben we snel helderheid!

**Comment by mellewynia** [comment]
@alexdekok , krijg net bericht van Marco ( @mdraijer) en streaming van de video content zou geen probleem moeten zijn! Een toegankelijk player is het volgende punt om uit te zoeken. Ik leg het alvast voor in Slack

**Comment by alexdekok** [comment]
@mellewynia super! check dan de player die ik zojuist deelde met de slackgroep svp.

**Comment by bwbroersma** [comment]
Vimeo is not the worst, YouTube (even with nocookie) is a GDPR violation without consent, I'm not sure about Vimeo.

@EdoPlantinga in all cases, one would use the native video tag, so `<video>`, with e.g. [MediaElementPlayer](http://www.mediaelementjs.com/) (mejs) or [video.js](https://github.com/videojs/video.js/) as 'add on', see [this minimal](https://github.com/mediaelement/mediaelement/blob/master/docs/usage.md#automatic-start) example how the 'add on' works on top of the native tags (feels like I'm [repeating](https://github.com/minvws/nl-covid19-notification-app-website/issues/61#issuecomment-652463721) myself :wink:).

**Comment by arianvp** [comment]
Your browser's HTML tag has native support for subtitles through the `<track>` element

Simply use the `<video>` tag and add `<track>` elements for each closed caption and use `video.js` to stub for IE11

Next to a closed captions you could add a Transcript. 

Example video element from duo.nl with captions + transcript:

https://nl-design-system.gitlab.io/nl-design-system/componenten/videos/index.html 

Next to this, you could consider adding sign language to the video.  

See https://www.w3.org/WAI/media/av/ for guidelines

Also see https://www.bundesregierung.de/breg-de/themen/corona-warn-app for the video of our german colleagues

**Comment by anneke** [comment]
Hiervoor staat ondertussen een pull-request open, deze wordt waarschijnlijk deze week afgerond. De media-player die hierin toegevoegd wordt aan de mogelijkheden op de site is in feite overgenomen van die op rijksoverheid.nl begreep ik van @mellewynia . Op dit moment wachten we nog op die content. Dit is een vrij oude issue van het Github Project bord, en zal ik sluiten zodra de media-player branch #140 gemerged is.
